.github/workflows/check-release.yml

```diff
@@ -0,0 +1,27 @@
+name: check-release
+on:
+  schedule:
+    - cron: "0 8 * * *"
+  push:
+    branches:
+      - "ensure-release"
+jobs:
+  check-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+        with:
+          fetch-depth: 0
+
+      - name: Check latest version in CHANGELOG.md has a corresponding release
+        run: |
+          changelog_version=$(grep -Eo 'v[0-9]+\.[0-9]+\.[0-9]+' CHANGELOG.md | head -n1)
+          echo "Top version found in CHANGELOG.md: $changelog_version"
+          release_version=$(curl https://api.github.com/repos/ios-xr/xrd-tools/releases/latest -s | jq .tag_name | tr -d '"')
+          echo "Latest release version from 'https://api.github.com/repos/ios-xr/xrd-tools/releases/latest': $release_version"
+          if [[ $release_version != $changelog_version ]]; then
+              echo "No release for $changelog_version (found in CHANGELOG.md), latest is $release_version" >&2
```

Comment:
Not sure it would result in an email - this is something we have planned to set up separately (I've added a note about this on the task). Could be shown on the dashboard, e.g. use the [GH API](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#get-a-workflow).

Reply:
Let's create a follow up task to make it more noticeable if this fails (e.g. email someone or put it on the dashboard), but happy to approve this for now.